Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -191,5 +191,15 @@ final class ConfigGenerator: ConfigGenerating {
                 }
             }
         }
+
+        if let deploymentTarget = target.deploymentTarget {
+            switch deploymentTarget {
+            case let .iOS(version, devices):
+                settings["TARGETED_DEVICE_FAMILY"] = .string(devices.map { "\($0.rawValue)" }.joined(separator: ","))
```

Comment:
A few tweaks needed here:

Once the devices is no longer an array:

```swift
var deviceFamilyValues: [Int] = []
if devices.contains(.iphone) {
  deviceFamilyValues.append(1)
}

if devices.contains(.ipad) {
  deviceFamilyValues.append(2)
}
```

For Mac it's slightly different, checking the box appears to set the following build settings:

```
SUPPORTS_MACCATALYST = YES
DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER = YES
``` 

(It doesn't map to a different device family)

Reply:
> For Mac it's slightly different, checking the box appears to set the following build settings:

let me double check this, haven't upgraded to Catalina yet 😅

Reply:
TODO: Add unit tests when this comment is resolved 👍

Reply:
> let me double check this, haven't upgraded to Catalina yet 😅

Hey @kwridan 👋 I am running my macbook on Catalina 👍 

> For Mac it's slightly different, checking the box appears to set the following build settings

I checked it 

<img width="465" alt="Screen Shot 2019-10-15 at 21 03 58" src="https://user-images.githubusercontent.com/10795657/66861554-860ef600-ef8f-11e9-89de-eece6021c2e5.png">

and I can confirm that indeed 

```
SUPPORTS_MACCATALYST = YES
DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER = YES
```

are added to build settings 👍 However if you do not have `Mac (requires macOS 10.15)` box checked, there's still 

```
SUPPORTS_MACCATALYST = NO
```

in a build settings.

Reply:
I pushed the newest changes to support that 🙇 `Support mac device for iOS deployment target`